components/brave_ads/browser/ads_service_impl.cc

```diff
@@ -379,12 +380,36 @@ void AdsServiceImpl::OnWalletUpdated() {
     return;
   }
 
-  const std::string payment_id =
-      profile_->GetPrefs()->GetString(brave_rewards::prefs::kPaymentId);
-  const std::string recovery_seed_base64 =
-      profile_->GetPrefs()->GetString(brave_rewards::prefs::kRecoverySeed);
+  const std::string wallet_string = profile_->GetPrefs()->GetString(
+      brave_rewards::prefs::kWalletBrave);
 
-  bat_ads_->OnWalletUpdated(payment_id, recovery_seed_base64);
+  if (wallet_string.empty()) {
+    return;
+  }
+
+  base::Optional<base::Value> value = base::JSONReader::Read(wallet_string);
+  if (!value || !value->is_dict()) {
+    VLOG(0) << "Parsing of brave wallet failed";
```

Comment:
nit: For consistency in ads logging could we please change to `Failed to parse wallet`